staging/src/k8s.io/legacy-cloud-providers/azure/azure_instances.go

```diff
@@ -239,80 +239,6 @@ func (az *Cloud) InstanceShutdownByProviderID(ctx context.Context, providerID st
 	return status == vmPowerStateStopped || status == vmPowerStateDeallocated || status == vmPowerStateDeallocating, nil
 }
 
-// InstanceMetadataByProviderID returns metadata of the specified instance.
-// InstanceMetadataByProviderID is part of InstancesV2 interface and is only used in cloud node-controller.
-func (az *Cloud) InstanceMetadataByProviderID(ctx context.Context, providerID string) (*cloudprovider.InstanceMetadata, error) {
```

Comment:
Safe to remove since InstanceV2 is disabled: https://github.com/kubernetes/kubernetes/blob/d3c1e81d5e6796db9389cd73073f48d6a37c4621/staging/src/k8s.io/legacy-cloud-providers/azure/azure.go#L667-L670